bigtop_toolchain/manifests/grpc.pp

```diff
@@ -0,0 +1,45 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class bigtop_toolchain::grpc {
+
+  require bigtop_toolchain::packages
+  $grpc_version = '1.28.0'
+  $proto_version = '3.5.1.1'
+  $proto_home = "/usr/local/protobuf-3.5.1.1"
+
+  if ($architecture == 'ppc64le') {
+
+  exec { "download grpc-java ${grpc_version}":
+      cwd  => "/usr/src",
+      command => "/usr/bin/wget https://github.com/grpc/grpc-java/archive/refs/tags/v${grpc_version}.tar.gz && mkdir -p grpc-java-${grpc_version} && /bin/tar -xvzf v${grpc_version}.tar.gz -C grpc-java-${grpc_version} --strip-components=1",
+      creates => "/usr/src/grpc-java-${grpc_version}",
+  }
+
+    file { "/usr/src/grpc-java-${grpc_version}/grpc-java-${grpc_version}-add-support-for-ppc64le.patch":
+      source => "puppet:///modules/bigtop_toolchain/grpc-java-${grpc_version}-add-support-for-ppc64le.patch",
+      require => Exec["download grpc-java ${grpc_version}"],
+    }
+
+  exec { "build grpc-java ${grpc_version}":
+      cwd => "/usr/src/grpc-java-${grpc_version}",
+      command => "/usr/bin/patch -p1 < grpc-java-${grpc_version}-add-support-for-ppc64le.patch && export LDFLAGS=-L/${proto_home}/lib && export CXXFLAGS=-I/${proto_home}/include && export LD_LIBRARY_PATH=/${proto_home}/lib && cd compiler && ../gradlew java_pluginExecutable -PskipAndroid=true",
```

Comment:
I think setting environment variables in the `environment` attribute is better for code simplicity.
https://github.com/sekikn/bigtop/commit/b94539c280df2b698a9d55db0cca753634d2d1b1#diff-f2963d87c19d71d34863fb0039c7b3d3d0e3699b77bfd7bd57009ca568f51da6R78

Reply:
Thanks, I am not a PP expert. My first time, I used https://github.com/apache/bigtop/blob/master/bigtop_toolchain/manifests/gradle.pp as a guide. Let me know if you need me to change it.

Reply:
oh got it:
environment => ['CXXFLAGS=-I/usr/local/protobuf-3.5.1.1/include', 'LDFLAGS=-L/usr/local/protobuf-3.5.1.1/lib'],

Reply:
Hi @sekikn, were you able to try to run the PR with all distros? If not, let me know. I'll try to do it over the weekend. Thanks

Reply:
@afsanjar Sorry, I couldn't have enough time to do it. It's really helpful if you could try it on this weekend!